### PR TITLE
PLT-7086 Improve error reporting for incorrect --match-asset-id values

### PR DESCRIPTION
**Pull request by neilmayhew**
Pre-submit checklist:
- Branch
    - [x] Tests are provided (if possible)
    - [x] Commit sequence broadly makes sense and have useful messages
    - [ ] Important changes are reflected in changelog.d of the affected packages
    - [x] Relevant tickets are mentioned in commit messages
- PR
    - [ ] (For external contributions) Corresponding issue exists and is linked in the description
    - [x] Targeting main unless this is a cherry-pick backport
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [ ] If relevant, reference the ADR in the PR and reference the PR in the ADR
    - [x] Reviewer requested

### CODE REVIEW

**Comment by neilmayhew** [comment]
The goofy error message is ultimately coming from `cardano-api` and should read "Unable" rather than "Enable". This typo occurs in two places. Is it OK if I submit a PR to fix the typo there?

This PR for `marconi` will still be needed, though, because we don't want the error message to mention `ScriptHash`.

**Comment by koslambrou** [comment]
> The goofy error message is ultimately coming from cardano-api and should read "Unable" rather than "Enable". This typo occurs in two places. Is it OK if I submit a PR to fix the typo there?

Yes! Definitely encouraged to do that

File: marconi-chain-index/src/Marconi/ChainIndex/CLI.hs [hunk 1]
```diff
@@ -317,17 +318,28 @@ commonMaybeTargetAssetParser =
...
-      parseAssetName = either (fail . show) pure . C.deserialiseFromRawBytesHex C.AsAssetName . Text.encodeUtf8
+      parseAssetName =
+        either (fail . C.displayError) pure . C.deserialiseFromRawBytesHex C.AsAssetName . Text.encodeUtf8
 
       parsePolicyId :: Text -> Opt.ReadM C.PolicyId
-      parsePolicyId = either (fail . show) pure . C.deserialiseFromRawBytesHex C.AsPolicyId . Text.encodeUtf8
+      parsePolicyId =
+        either (fail . displayError') pure . C.deserialiseFromRawBytesHex C.AsPolicyId . Text.encodeUtf8
+
+      displayError' =
```
Comment (koslambrou) [inline marconi-chain-index/src/Marconi/ChainIndex/CLI.hs:328]: Add a comment on why we need this vs just using `displayError`. You can just say like in your previous comment that we don't want to mention `ScriptHash` in the error message.

**Review by koslambrou — APPROVED** [review]
LGTM. Good work :)

**Pull request merged**